deps/curl-7.88.1/lib/vtls/sf_ocsp.c

```diff
@@ -266,6 +268,53 @@ static const unsigned int MAX_SLEEP_TIME = 16000000;
 static const int MAX_RETRY = 1;
 
 
+/* Encode request data as a part of URL
+ * Encode all characters reserved by URL
+ * @param url_data The source data needs to be encoded_ptr
+ * @param data_size The size of the source data
+ * @param encoded_ptr The allocated buffer filled with encoded data, need to be freed by caller
+ * return the size of encoded data
+ */
+size_t encodeUrlData(const char *url_data, size_t data_size, char** encoded_ptr)
+{
+  // allocate buffer 3 times larger than source data in case every single, add 1 for '\0'
+  // character needs to be encoded as %xx
+  size_t buf_len = data_size * 3 + 1;
+  char* encode_buf = (char*)malloc(buf_len);
```

Comment:
Make sure malloc is right API here and during free it calls curl_free.

Reply:
Confirmed, following existing curl implementation.